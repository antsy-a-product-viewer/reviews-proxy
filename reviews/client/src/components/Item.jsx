import React from 'react';
import itemStyles from './css/itemStyles.css';

const Item = (props) => {
  const clickItem = (event) => {
    props.visitItem(event.target.id);
  };

  if (props.showPrice === 'false') {
    return (
      <div style={itemStyles.wrapper}>
        <img id={props.review.item_id} style={itemStyles.itemImg} onClick={clickItem} src={props.review.item_img}></img>
        <div id={props.review.item_id} style={itemStyles.reviewItemInfo} onClick={clickItem}>{props.review.item_name}</div>
      </div>
    );
  } else {
    return (
      <div style={itemStyles.wrapper}>
        <img id={props.review.item_id} style={itemStyles.itemImg} onClick={clickItem} src={props.review.item_img}></img>
        <div style={itemStyles.reviewItemInfo}>
          <div id={props.review.item_id} onClick={clickItem}>{props.review.item_name}</div>
          <div>${props.review.price}.00</div>
        </div>
      </div>
    );
  }
};

export default Item;
