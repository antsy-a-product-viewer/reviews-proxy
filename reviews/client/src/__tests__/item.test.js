import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import Item from '../components/Item.jsx';

configure({ adapter: new Adapter() });

it('should render price with if props.price is set to true', () => {
  const review = {
    'review_id': 119,
    'user_img': 'https://s3-us-west-1.amazonaws.com/antsyuserimages/00027.jpg',
    'user_name': 'Zella Blick',
    'created_at': '2014-03-14T22:37:45.000Z',
    'stars': 4,
    'review': 'Try to hack the SQL transmitter, maybe it will navigate the multi-byte bus!',
    'item_id': 30,
    'item_img': 'https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg',
    'item_name': 'Handmade Concrete Salad'
  };
  
  const props = {
    showPrice: 'true',
    review: review
  };

  const item = shallow(<Item key={props.review.item_id} showPrice={props.showPrice} review={props.review}/>);
  
  expect(item).toMatchSnapshot();
});