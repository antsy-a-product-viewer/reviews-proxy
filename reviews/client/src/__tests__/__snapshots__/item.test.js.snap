// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render price with if props.price is set to true 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginTop": 10,
    }
  }
>
  <img
    id={30}
    onClick={[Function]}
    src="https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg"
    style={
      Object {
        "alignSelf": "center",
        "cursor": "pointer",
        "height": 75,
        "width": 75,
      }
    }
  />
  <div
    id={30}
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "margin": "25px",
      }
    }
  >
    <div>
      Handmade Concrete Salad
    </div>
    <div>
      $
      .00
    </div>
  </div>
</div>
`;
